### PR TITLE
Created uploading route

### PR DESCRIPTION
**Pull request by Souvikns**
I am currently thinking of a better way for routing, This is not final but, rather a first approach, to solve the problem.  

closing #11

### CODE REVIEW

File: controller/file.js [hunk 1]
```diff
@@ -1,26 +1,30 @@
 const AWS = require('aws-sdk')
+const File = require('../model/file')
+const { v4: uuid4, v4 } = require('uuid')
```
Comment (aggarwaldev) [inline controller/file.js:3]: don't import same module twice. Recommend you to use `{v4:uuid4}` only

File: controller/file.js [hunk 2]
```diff
@@ -1,26 +1,30 @@
...
     accessKeyId: process.env.AWS_ACCESS_KEY_ID,
     secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY
 })
 
-const uploadSignleFile = async (req, res) => {
+const uploadSignleFile = async (file) => {
     // Upload the file to s3 bucket
-    let myFile = req.file.originalname.split('.')
+    let myFile = file.originalname.split('.')
     let fileType = myFile[myFile.length - 1]
```
Comment (aggarwaldev) [inline controller/file.js:12]: Use `file.mimetype` for detecting fileType.
Refer to https://www.npmjs.com/package/multer#file-information

File: controller/file.js [hunk 3]
```diff
@@ -1,26 +1,30 @@
...
+    let key = v4()
```
Comment (aggarwaldev) [inline controller/file.js:13]: Use `const` instead of `let`

File: controller/file.js [hunk 4]
```diff
@@ -31,16 +35,86 @@ const downloadFile = async (req, res) => {
...
+    }
+}
+
+const createDoc = async (req, res) => {
+    let { level, root } = req.params
+    let { name, type } = req.body
+    let doc = req.file
+
+    try {
+        switch (type) {
```
Comment (aggarwaldev) [inline controller/file.js:77]: recommend you to use `type.toUpperCase()` in switch statement

File: package.json [hunk 5]
```diff
@@ -30,7 +30,8 @@
     "multer": "^1.4.2",
     "passport": "^0.4.1",
     "passport-jwt": "^4.0.0",
-    "passport-local": "^1.0.0"
+    "passport-local": "^1.0.0",
+    "uuid": "^8.2.0"
```
Comment (aggarwaldev) [inline package.json:34]: For future reference, split package installation and code into separate commits.

Comment (aggarwaldev) [inline package.json:34]: For future reference: Split package installation and code changes into separate commits

File: controller/file.js (hunk 1 above)
Comment (Souvikns) [reply controller/file.js:3]: Oh this was auto imported I guess,

**Comment by Souvikns** [comment]
Currently the routing design is very bad, and I think we should change that, next I am gonna change that, I guess I have to change the code for controller also to accommodate that.

**Pull request merged**